src/language/LanguageManager.js

```diff
@@ -398,7 +398,40 @@ define(function (require, exports, module) {
     function _resetPathLanguageOverrides() {
         _filePathToLanguageMap = {};
     }
+    
+    /**
+     * Get the file extension (excluding ".") given a path OR a bare filename.
+     * Returns "" for names with no extension.
+     * If the only `.` in the file is the first character,
+     * returns "" as this is not considered an extension.
+     * This method considers known extensions which include `.` in them.
+     *
+     * @param {string} fullPath full path to a file or directory
+     * @return {string} Returns the extension of a filename or empty string if
+     * the argument is a directory or a filename with no extension
+     */
+    function getCompoundFileExtension(fullPath) {
+        var baseName = FileUtils.getBaseName(fullPath),
```

Comment:
Funny I didn't catch this earlier. Here we have the proper relationship between LanguageManager and FileUtils (the former relies on the latter), but because of the deprecation warning, there is still a circular loop here but it is not crashing. Perhaps I should submit a quick followup PR to make LanguageManager load FileUtils in async until the deprecation warning is removed to prevent any possible crashes?